gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayWebAppResource.java

```diff
@@ -423,4 +436,67 @@ public Response readExactMatchSourceSelector()
         List<ResourceGroupsManager.ExactSelectorsDetail> selectorsDetailList = resourceGroupsManager.readExactMatchSourceSelector();
         return Response.ok(Result.ok(selectorsDetailList)).build();
     }
+
+    @GET
+    @RolesAllowed("USER")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/getRoutingRules")
+    public Response readRoutingRules()
+    {
+        String content = null;
+        try {
+            String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+            content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)));
+            ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            List<RoutingRules> routingRulesList = new ArrayList<>();
+            while (parser.nextToken() != null) {
+                RoutingRules routingRules = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRules);
+            }
+            return Response.ok(Result.ok(routingRulesList)).build();
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @POST
+    @RolesAllowed("ADMIN")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/updateRoutingRules")
+    public Response updateRoutingRules(RoutingRules routingRules)
+    {
+        String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+        ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+        List<RoutingRules> routingRulesList = new ArrayList<>();
+
+        try {
+            String content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)));
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            while (parser.nextToken() != null) {
+                RoutingRules routingRule = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRule);
+            }
+
+            for (int i = 0; i < routingRulesList.size(); i++) {
+                if (routingRulesList.get(i).name().equals(routingRules.name())) {
+                    routingRulesList.set(i, routingRules);
+                    break;
+                }
+            }
```

Comment:
as the name of this method suggests, this updates an existing rule. It does not appear that this change adds support for deleting an existing rule or adding a new one. I guess this may be due to the UI changes required, but would it be difficult to add this functionality?

Reply:
I don't think it would be difficult. I thought of initially going with only updates and in a separate PR i will put Add and Delete functionality. Will that be ok? Or should I add it in this PR itself?